core/common/src/main/java/tachyon/wire/WorkerInfo.java

```diff
@@ -0,0 +1,218 @@
+   */
+  public WorkerInfo(tachyon.thrift.WorkerInfo workerInfo) {
+    mId = workerInfo.getId();
+    mAddress = new WorkerNetAddress(workerInfo.getAddress());
+    mLastContactSec = workerInfo.getLastContactSec();
+    mState = workerInfo.getState();
+    mCapacityBytes = workerInfo.getCapacityBytes();
+    mUsedBytes = workerInfo.getUsedBytes();
+    mStartTimeMs = workerInfo.getStartTimeMs();
+  }
+
+  /**
+   * @return the worker id
+   */
+  public long getId() {
+    return mId;
+  }
+
+  /**
+   * @return the worker address
+   */
+  public WorkerNetAddress getAddress() {
+    return mAddress;
+  }
+
+  /**
+   * @return the worker last contact (in seconds)
+   */
+  public int getLastContactSec() {
+    return mLastContactSec;
+  }
+
+  /**
+   * @return the worker state
+   */
+  public String getState() {
+    return mState;
+  }
+
+  /**
+   * @return the worker total capacity (in bytes)
+   */
+  public long getCapacityBytes() {
+    return mCapacityBytes;
+  }
+
+  /**
+   * @return the worker used capacity (in bytes)
+   */
+  public long getUsedBytes() {
+    return mUsedBytes;
+  }
+
+  /**
+   * @return the worker start time (in milliseconds)
+   */
+  public long getStartTimeMs() {
+    return mStartTimeMs;
+  }
+
+  /**
+   * @param id the worker id to use
+   * @return the worker descriptor
+   */
+  public WorkerInfo setId(long id) {
+    mId = id;
+    return this;
+  }
+
+  /**
+   * @param address the worker address to use
+   * @return the worker descriptor
+   */
+  public WorkerInfo setAddress(WorkerNetAddress address) {
+    Preconditions.checkNotNull(address);
+    mAddress = address;
+    return this;
+  }
+
+  /**
+   * @param lastContactSec the worker last contact (in seconds) to use
+   * @return the worker descriptor
+   */
+  public WorkerInfo setLastContactSec(int lastContactSec) {
+    mLastContactSec = lastContactSec;
+    return this;
+  }
+
+  /**
+   * @param state the worker state to use
+   * @return the worker descriptor
+   */
+  public WorkerInfo setState(String state) {
+    Preconditions.checkNotNull(state);
+    mState = state;
+    return this;
+  }
+
+  /**
+   * @param capacityBytes the worker total capacity (in bytes) to use
+   * @return the worker descriptor
+   */
+  public WorkerInfo setCapacityBytes(long capacityBytes) {
+    mCapacityBytes = capacityBytes;
+    return this;
+  }
+
+  /**
+   * @param usedBytes the worker used capacity (in bytes) to use
+   * @return the worker descriptor
+   */
+  public WorkerInfo setUsedBytes(long usedBytes) {
+    mUsedBytes = usedBytes;
+    return this;
+  }
+
+  /**
+   * @param startTimeMs the worker start time (in milliseconds) to use
+   * @return the worker descriptor
+   */
+  public WorkerInfo setStartTimeMs(long startTimeMs) {
+    mStartTimeMs = startTimeMs;
+    return this;
+  }
+
+  /**
+   * @return thrift representation of the worker descriptor
+   */
+  public tachyon.thrift.WorkerInfo toThrift() {
```

Comment:
What do you think of implementing an interface with a `toThrift` method for all the wire classes?